test/testcmdlineparser.cpp

```diff
@@ -368,21 +370,36 @@ class TestCmdlineParser : public TestFixture {
         ASSERT_EQUALS("", GET_REDIRECT_OUTPUT);
     }
 
-    void showversion() {
+    void version() {
         REDIRECT;
         const char * const argv[] = {"cppcheck", "--version"};
         ASSERT(parser->parseFromArgs(2, argv));
-        ASSERT_EQUALS(true, parser->getShowVersion());
-        ASSERT_EQUALS("", logger->str()); // version is not actually shown
+        ASSERT_EQUALS("Cppcheck 2.13 dev\n", logger->str());
+        ASSERT_EQUALS("", GET_REDIRECT_OUTPUT);
     }
 
+    void versionWithCfg() {
+        REDIRECT;
+        ScopedFile file("cppcheck.cfg",
+                "{\n"
+                "\"productName\": \"The Product\""
+                "}\n");
+        const char * const argv[] = {"cppcheck", "--version"};
+        ASSERT(parser->parseFromArgs(2, argv));
+        // TODO: somehow the config is not loaded on some systems
```

Comment:
I will look into this in the PR which adds proper error reporting to the config loading function.